// @react-keycloak/ssr/lib-commonjs/persistors/server.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SSRCookies = exports.ExpressCookies = void 0;
var js_base64_1 = require("js-base64");
var utils_1 = require("./utils");
exports.ExpressCookies = function (req) { return ({
    setTokens: function () { },
    getTokens: function () {
        var _a = req.cookies || {}, kcIdToken = _a.kcIdToken, kcToken = _a.kcToken;
        return {
            idToken: kcIdToken ? js_base64_1.Base64.decode(kcIdToken) : undefined,
            token: kcToken ? js_base64_1.Base64.decode(kcToken) : undefined,
        };
    },
    resetTokens: function () { },
}); };
exports.SSRCookies = function (cookies) { return ({
    setTokens: function (_a) {
        var idToken = _a.idToken, token = _a.token;
        !!token && utils_1.setCookie('kcToken', utils_1.encode(token));
        !!idToken && utils_1.setCookie('kcIdToken', utils_1.encode(idToken));
    },
    getTokens: function () {
        var tknStr = utils_1.getCookie('kcToken', cookies);
        var idTknStr = utils_1.getCookie('kcIdToken', cookies);
        return {
            idToken: idTknStr ? utils_1.decode(idTknStr) : '',
            refreshToken: '',
            token: tknStr ? utils_1.decode(tknStr) : '',
        };
    },
    resetTokens: function () {
        utils_1.removeCookie('kcToken');
        utils_1.removeCookie('kcIdToken');
    },
}); };
