// @react-keycloak/ssr/lib-commonjs/internals/keycloak.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getKeycloakInstance = exports.getKeycloakStub = void 0;
var utils_1 = require("./utils");
// Keycloak singleton
var keycloakInstance;
// KeycloakStub singleton
var keycloakStubInstance;
// this is a fake Keycloak instance we use to initialize Keycloak on the server.
// This gets over-written as soon as Keycloak is initialized on the client.
exports.getKeycloakStub = function (persistor) {
    var kcTokens = persistor.getTokens();
    keycloakStubInstance = {
        init: function () { return Promise.resolve(true); },
        updateToken: function () { return Promise.resolve(false); },
        idToken: kcTokens.idToken,
        token: kcTokens.token,
        refreshToken: kcTokens.refreshToken,
        authenticated: !!kcTokens.idToken && !!kcTokens.token,
    };
    return keycloakStubInstance;
};
var Keycloak = !utils_1.isServer() ? require('keycloak-js') : null;
exports.getKeycloakInstance = function (keycloakConfig, persistor, recreate) {
    if (recreate === void 0) { recreate = false; }
    var isServerCheck = utils_1.isServer();
    if (recreate || (!keycloakInstance && !isServerCheck)) {
        keycloakInstance = new Keycloak(keycloakConfig);
    }
    return !isServerCheck ? keycloakInstance : exports.getKeycloakStub(persistor);
};
